src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -1007,8 +1010,17 @@ var ReactDOMFiberComponent = {
               nextProp,
             );
           } else {
-            // $FlowFixMe - Should be inferred as not undefined.
-            extraAttributeNames.delete(propKey.toLowerCase());
+            let ownNamespace = parentNamespace;
+            if (ownNamespace === HTML_NAMESPACE) {
+              ownNamespace = getIntrinsicNamespace(tag);
+            }
+            if (ownNamespace === HTML_NAMESPACE) {
+              // $FlowFixMe - Should be inferred as not undefined.
+              extraAttributeNames.delete(propKey.toLowerCase());
```

Comment:
I wonder if it's worth adding a comment for why we need to downcase the propKey (because HTML reports lowercase, right?)